### PR TITLE
Ft:158491860 View all User Feedback

### PR DESCRIPTION
**Pull request by andela-fojuri**
## What does this PR do?
- Displays all user feedback from the backend
## Description of Task to be completed?
- All User Feedback should displayed when I click on User Feedback Link

## What are the relevant pivotal tracker stories?
[#158491860](https://www.pivotaltracker.com/story/show/158491860)

## Related PRs branch | PR (branch_name) | (pr_link)
Branch | PR
------ | ------
other_pr_production | [link]()
other_pr_master | [link]()
## Todos
- [x] Raise PR
- [ ] Test
- [ ] Documentation
## Screenshots

### CODE REVIEW

File: src/components/UserFeedbackComponent.jsx [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  constructor() {
+    super();
+    this.state = {
+      activePage: 1,
+      limit: 10,
+    };
+  }
+
+  componentDidMount() {
+    this.props.feedbackAction(this.state.defaultPage);
```
Comment (theterminalguy) [inline src/components/UserFeedbackComponent.jsx]: Where is `this.state.defaultPage` defined ?

File: src/components/UserFeedbackComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+  loadFeedback = () => {
+    if (this.emptyFeedbackCount()) {
+      return (
+        <Table.Row>
+          <Table.Cell colSpan="6">No Data found</Table.Cell>
+        </Table.Row>
+      );
+    }
+    const feedbacks = this.props.feedback.map(feedback => (
```
Comment (theterminalguy) [inline src/components/UserFeedbackComponent.jsx]: You can't pluralize **feedback** either change the name of the variable to something like **feedback_records** or just call it feedback. There is no such word as feedbacks.

**Comment by theterminalguy** [comment]
Also, @andela-fojuri change the name of the reducer from `feedback` to `user_feedback`

File: src/_reducers/feedback.reducer.js [hunk 3]
```diff
@@ -3,24 +3,24 @@ import constants from '../_constants';
 const { LOAD_FEEDBACK_SUCCESS, LOAD_FEEDBACK_FAILURE } = constants;
 
 const initialState = {
-    feedback: [],
-    feedbackCount: 0,
-}
+  feedback: [],
+  feedbackCount: 0,
```
Comment (theterminalguy) [inline src/_reducers/feedback.reducer.js]: Is there a reason you are storing the `feedbackCount` in the reducer? Is this something 

1. used in pagination,
2. required by the backend
3. returned by the backend

Let me know

File: src/components/UserFeedbackComponent.jsx (hunk 2 above)
Comment (andela-fojuri) [reply src/components/UserFeedbackComponent.jsx]: Yes, I agree.. This was already added before I added my changes.

File: src/_reducers/feedback.reducer.js (hunk 3 above)
Comment (andela-fojuri) [reply src/_reducers/feedback.reducer.js]: It was already added before I raised this PR. I can remove it if it's not needed.

File: src/components/UserFeedbackComponent.jsx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+  }
+
+  handlePaginationChange = (e, { activePage }) => {
+    this.setState({ activePage });
+    this.props.feedbackAction(activePage);
+  }
+
+  handlePageTotal = () => Math.ceil(this.props.feedbackCount / this.state.limit)
+
+  emptyFeedbackCount = () => (this.props.feedbackCount === 0)
```
Comment (theterminalguy) [inline src/components/UserFeedbackComponent.jsx]: There is a few issues with this method 

1. `emptyFeedbackCount` sounds like you are counting emptyFeedback which seems like a domain term. A domain term is a word a business or organization as chosen to mean something so everyone can understand what they are trying to communicate

2. Is `this.props.feedbackCount` in sync with `this.props.feedback.lenght` ?

3. Change the method name to something better. Let me know what name you come up with

File: src/_reducers/feedback.reducer.js (hunk 3 above)
Comment (theterminalguy) [reply src/_reducers/feedback.reducer.js]: @andela-fojuri you still haven't answered the questions

Comment (andela-fojuri) [reply src/_reducers/feedback.reducer.js]: I'm not sure of the main reason the person that worked on this was storing the feedbackCount in the reducer. It's being used in pagination but not returned/required by the backend.

**Pull request merged**